src/pages/sign/signIn/page.tsx

```diff
@@ -1,11 +1,82 @@
-import { Layout } from "antd";
+import { Button, Form, Input, Layout } from "antd";
+import { useUnit } from "effector-react";
 import { PageHeader } from "../../../shared/ui/PageHeader/PageHeader";
+import { users } from "./mocks";
+import { signInAdmin, signInUser } from "./model";
+import "./page.css";
 
 const { Content } = Layout;
 
-export const SignInPage = () => (
-  <Layout>
-    <PageHeader title="Вход" />
-    <Content />
-  </Layout>
-);
+type Value = {
+  nickName: string;
+  password: string;
+};
+
+export const SignInPage = () => {
+  const [signInAdminFn, signInUserFn] = useUnit([signInAdmin, signInUser]);
+
+  const handleFinish = (user: Value) =>
+    user.nickName === "admin" ? signInAdminFn() : signInUserFn();
+
+  return (
+    <Layout>
+      <PageHeader title="Вход" />
+      <Content className="Content--signInPage">
+        <Form
+          className="Form"
+          wrapperCol={{ span: 6, offset: 9 }}
+          onFinish={handleFinish}
+          autoComplete="off"
+          validateTrigger="onFinish"
+        >
+          <Form.Item
+            name="nickName"
+            rules={[
+              { required: true, message: "Введите ник!" },
+              () => ({
+                validator(_, value) {
+                  if (users.some((user) => user.nickName === value)) {
+                    return Promise.resolve();
+                  }
+                  return Promise.reject(
+                    new Error("Пользователь не существует")
+                  );
+                },
+              }),
+            ]}
+          >
+            <Input placeholder="ник" allowClear />
+          </Form.Item>
+          <Form.Item
+            name="password"
+            rules={[
+              { required: true, message: "Введите пароль!" },
+              ({ getFieldValue }) => ({
```

Comment:
I think you can try to take out this function separately, so that it looks simpler in the component itself.
Is it possible to put this logic into an effector?